### PR TITLE
UX: exclude irrelevant search filters for anonymous users

### PR DESCRIPTION
**Pull request by xrav3nz**
> This PR tries to resolve https://meta.discourse.org/t/hide-i-liked-ive-read-etc-from-advanced-search-for-anonymous-users/64816.

---

On the advanced search page, filters like "I've read", "I'm watching", etc, are irrelevant to anonymous users and should be hidden.

Open to suggestions, cheers!

---

**For authorized users:**
authorized_user

**For anonymous users:**
anonymous_user

### CODE REVIEW

File: test/javascripts/acceptance/search-full-test.js.es6 [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { acceptance, waitFor } from "helpers/qunit-helpers";
 acceptance("Search - Full Page", {
   settings: {tagging_enabled: true},
+  loggedIn: true,
```
Comment (tgxworld) [inline test/javascripts/acceptance/search-full-test.js.es6]: Can we add a test here for non-logged in users?

**Comment by tgxworld** [comment]
Generally this PR looks good to me/

File: test/javascripts/acceptance/search-test.js.es6 [hunk 2]
```diff
@@ -73,3 +73,53 @@ QUnit.test("Search with context", assert => {
...
+  visit("/search?expanded=true");
+
+  andThen(() => {
+    assert.ok(exists('select#in option[value=unseen]'));
+    assert.ok(exists('select#in option[value=posted]'));
+    assert.ok(exists('select#in option[value=watching]'));
+    assert.ok(exists('select#in option[value=tracking]'));
+    assert.ok(exists('select#in option[value=bookmarks]'));
+  });
+});
```
Comment (xrav3nz) [inline test/javascripts/acceptance/search-test.js.es6]: The tests may seem a bit redundant. However, IMO, `exists` and `notExists` are both needed to make sure the elements are only rendered differently due to logging in. Open to suggestions on ways to improve testing tho.

File: test/javascripts/acceptance/search-test.js.es6 [hunk 3]
```diff
@@ -73,3 +73,53 @@ QUnit.test("Search with context", assert => {
...
+  visit("/search?expanded=true");
+
+  andThen(() => {
+    assert.ok(exists('.search-advanced-options .in-likes'));
+    assert.ok(exists('.search-advanced-options .in-private'));
+    assert.ok(exists('.search-advanced-options .in-seen'));
+  });
+});
+
+QUnit.test(`"I've not read", "I posted in", "I'm watching", "I'm tracking",
```
Comment (tgxworld) [inline test/javascripts/acceptance/search-test.js.es6]: I would just combine these tests with the one above.

Probably just two description for the tests:

`Quni.test("right filters are shown to anonymous users")` and `Quni.test("right filters are shown to logged in users")`

File: test/javascripts/acceptance/search-test.js.es6 (hunk 2 above)
Comment (tgxworld) [inline test/javascripts/acceptance/search-test.js.es6]: Yup I agree with you that we'll need `exists` and `notExists`

File: test/javascripts/acceptance/search-test.js.es6 [hunk 4]
```diff
@@ -73,3 +73,49 @@ QUnit.test("Search with context", assert => {
...
+
+QUnit.test("Right filters are shown to anonymous users", assert => {
+  visit("/search?expanded=true");
+
+  andThen(() => {
+    assert.ok(exists('select#in option[value=first]'));
+    assert.ok(exists('select#in option[value=pinned]'));
+    assert.ok(exists('select#in option[value=unpinned]'));
+    assert.ok(exists('select#in option[value=wiki]'));
+    assert.ok(exists('select#in option[value=images]'));
```
Comment (xrav3nz) [inline test/javascripts/acceptance/search-test.js.es6]: Took the liberty to include the filters from [`inOptionsForAll`](https://github.com/discourse/discourse/pull/5032/files#diff-d8f8a0225b6661e2dd68852bdeb0b9e5R40); making sure they are rendered at all times.

**Comment by tgxworld** [comment]
@xrav3nz Thank you so much :+1: It was a pleasure reviewing your PR.

**Pull request merged**